src/codeMarker.ts

```diff
@@ -551,6 +563,66 @@ export class CodeMarker implements vscode.TreeDataProvider<TreeEntry> {
         this.refresh(uri);
     }
 
+    addPartialAudited(): void {
+        const editor = vscode.window.activeTextEditor;
+        if (editor === undefined) {
+            return;
+        }
+        const uri = editor.document.uri;
+        // get path relative to workspace
+        const relativePath = path.relative(this.workspacePath, uri.fsPath);
+
+        // check if file is already in list
+        const index = this.auditedFiles.findIndex((file) => file.path === relativePath);
+
+        // if file is already audited ignore
+        if (index > -1) {
+            return;
+        }
+
+        const location = this.getActiveSelectionLocation();
+        const exactMatchIndex = this.partialAuditedFiles.findIndex(
+            (file) => file.path === relativePath && file.location.startLine === location.startLine && file.location.endLine === location.endLine,
+        );
+
+        // this section is already marked. Remove it then
+        if (exactMatchIndex > -1) {
+            this.partialAuditedFiles.splice(exactMatchIndex, 1);
+        } else {
+            // allows to remove sections from the begining or end
+            const partialMatchIndex = this.partialAuditedFiles.findIndex(
+                (file) => file.path === relativePath && (file.location.startLine === location.startLine || file.location.endLine === location.endLine),
+            );
+            if (partialMatchIndex > -1) {
+                const partialMatch = this.partialAuditedFiles[partialMatchIndex];
+                if (partialMatch.location.startLine === location.startLine) {
+                    partialMatch.location.startLine = location.endLine + 1;
+                }
+                if (partialMatch.location.endLine === location.endLine) {
+                    partialMatch.location.endLine = location.startLine - 1;
+                }
+            } else {
+                this.partialAuditedFiles.push({
+                    path: relativePath,
+                    author: this.username,
+                    location: {
+                        description: "Partially audited",
+                        label: "Partial",
+                        path: relativePath,
+                        startLine: location.startLine,
+                        endLine: location.endLine,
+                    },
```

Comment:
My initial goal here was to reuse the `Location` type that is already in the code base. If you prefer having only the mentioned fields I can refactor the `Location` type to be the bare minimum and add new types for the different location use cases.
lmk what you think

Reply:
Let's leave the Location type since it is used by other vscode apps.
I would suggest changing the type to 
```
export interface PartiallyAuditedFile {
    path: string;
    author: string;
    startLine: number;
    endLine: number;
}
```

Reply:
done